test/validation/api/classification/odp_classification_basic.c

```diff
@@ -395,6 +395,47 @@ static void classification_test_pmr_composite_create(void)
 	odp_pktio_close(pktio);
 }
 
+static void classification_test_create_cos_with_hash_queues(void)
+{
+	odp_pool_t pool;
+	odp_cls_capability_t capa;
+	int ret;
+	odp_queue_param_t q_param;
+	odp_cls_cos_param_t cls_param;
+	odp_cos_t cos;
+
+	pool = pool_create("cls_basic_pool");
+	CU_ASSERT_FATAL(pool != ODP_POOL_INVALID);
+
+	ret = odp_cls_capability(&capa);
+	CU_ASSERT_FATAL(ret == 0);
+
+	odp_queue_param_init(&q_param);
+	odp_cls_cos_param_init(&cls_param);
+	cls_param.num_queue = capa.max_hash_queues;
+	cls_param.queue_param = q_param;
+	cls_param.hash_proto.all_bits = capa.hash_protocols.all_bits;
+	cls_param.pool = pool;
+
+	cos = odp_cls_cos_create(NULL, &cls_param);
+	CU_ASSERT_FATAL(cos != ODP_COS_INVALID);
+
+	ret = odp_cos_destroy(cos);
+	CU_ASSERT(ret == 0);
+
+	odp_pool_destroy(pool);
+}
+
+static int check_capa_cos_hashing(void)
+{
+	odp_cls_capability_t capa;
+
+	if (odp_cls_capability(&capa) < 0)
+		return 0;
+
+	return capa.max_hash_queues > 1 && capa.hash_protocols.all_bits;
```

Comment:
Perhaps `capa.hash_protocols.all_bits != 0` assert could be added to the test itself. It doesn't seem to make much sense to support multiple queues and zero hash protocols.

Reply:
True, it's a bug really. Asserted now in the test case in v2 (fatally as passing zero in cls params might lead to weird behavior in some implementations).